test/cnf_test.jl

```diff
@@ -111,9 +106,7 @@ function loss_adjoint(θ)
     return mean(@. -logpx + 0.1 * λ₁ + λ₂)
 end
 
-res = DiffEqFlux.sciml_train(loss_adjoint, 0.01f0 .* ffjord_test.p,
-                             ADAM(0.1), cb = cb,
-                             maxiters = 100)
+res = DiffEqFlux.sciml_train(loss_adjoint, 0.01f0 .* ffjord_test.p, ADAM(0.1), GalacticOptim.AutoZygote(), cb=cb, maxiters = 300) 
```

Comment:
after making sure tests pass, all of these sciml_train calls should be replaced with GalacticOptim code. The tests should all use what we say is the correct way to do it.

Reply:
So the GalacticOptim way of doing it has been added in galacticoptim.jl tests, I wanted to (and that's what has been discussed here and in previous PRs) to keep around `sciml_train` that passes it forward to GalacticOptim (with a warning) and then once we decide to completely drop support for it I think we can switch all the tests to the GalacticOptim interface.

P.S. The `sciml_train` wrapper for GalacticOptim is very user friendly

Reply:
>So the GalacticOptim way of doing it has been added in galacticoptim.jl tests, I wanted to (and that's what has been discussed here and in previous PRs) to keep around sciml_train that passes it forward to GalacticOptim (with a warning) and then once we decide to completely drop support for it I think we can switch all the tests to the GalacticOptim interface.

Yes, there should be a depwarn test, but we should have like one. Most of the tests should do what the tutorial does. (and the docs need an update, but that can come in a follow up)